src/controllers/dynakube/version/image_hash_test.go

```diff
@@ -49,3 +54,36 @@ func assertVersionStatusEquals(t *testing.T, registry *fakeRegistry, imageRef re
 	assert.NoError(t, err, "Image version is unexpectedly unknown for '%s'", imageRef.String())
 	assert.Contains(t, versionStatus.ImageID, expectedDigest.Digest)
 }
+
+func TestGetImageVersion(t *testing.T) {
+	t.Run("without proxy or trustedCAs", func(t *testing.T) {
+		registryMockClient := &mocks.MockClient{}
+		registryMockClient.On("GetImageVersion", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(registry.ImageVersion{}, nil)
+		imageName := "dynatrace-operator:1.0.0"
+		dockerConfig := dockerconfig.NewDockerConfig(fake.NewClient(), dynakube.DynaKube{})
+
+		got, err := GetImageVersion(context.TODO(), registryMockClient, imageName, dockerConfig)
+		assert.NotNil(t, got)
+		assert.Nil(t, err)
+	})
+	t.Run("with proxy", func(t *testing.T) {
+		registryMockClient := &mocks.MockClient{}
+		registryMockClient.On("GetImageVersion", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(registry.ImageVersion{}, nil)
+		imageName := "dynatrace-operator:1.0.0"
+		dockerConfig := dockerconfig.NewDockerConfig(fake.NewClient(), dynakube.DynaKube{Spec: dynakube.DynaKubeSpec{Proxy: &dynakube.DynaKubeProxy{Value: "dummy-proxy"}}})
+
+		got, err := GetImageVersion(context.TODO(), registryMockClient, imageName, dockerConfig)
+		assert.NotNil(t, got)
+		assert.Nil(t, err)
+	})
+	// t.Run("with trustedCAs", func(t *testing.T) {
```

Comment:
Since you're moking with client - you can remove this test.

Reply:
I rewrote the test using a hardcoded certificate as we discussed!

Reply:
make sense.